backend/corpora/common/entities/collection.py

```diff
@@ -150,6 +152,108 @@ def to_dict(db_object):
 
         return results
 
+    @classmethod
+    def list_collections_for_curator_api(cls, session: Session, visibility: str = None) -> typing.List[dict]:
+        """
+        Get a subset of columns, in dict form, for all Collections with the specified visibility. If visibility is None,
+        return *all* Collections.
+        @param session: the SQLAlchemy session
+        @param visibility: the CollectionVisibility string name
+        @return: a list of dict representations of Collections
+        """
+
+        def datetime_serializer(dt: datetime):
+            return mktime(dt.timetuple())
+
+        def enum_serializer(e: enum.Enum):
+            return e.name
+
+        def array_packager(o: object):
+            return o if type(o) == list else [o]
+
+        collection_columns = [
+            "id",
+            "name",
+            ("visibility", enum_serializer),
+            "tombstone",
+            "contact_name",
+            "contact_email",
+            "curator_name",
+            ("revised_at", datetime_serializer),
+            ("created_at", datetime_serializer),
+            "links",
+            "datasets",
+            ("published_at", datetime_serializer),
+            "description",
+            "publisher_metadata",
+            "revision_of",
+            "tombstone",
+            "owner",  # Needed for determining view permissions
+        ]
+
+        link_columns = [
+            "link_name",
+            "link_url",
+            ("link_type", enum_serializer),
+        ]
+
+        dataset_columns = [
+            "id",
+            "curator_tag",
+            "tissue",
+            "assay",
+            "disease",
+            ("organism", array_packager),
+            "tombstone",
+        ]
+
+        filters = []
+        if visibility == CollectionVisibility.PUBLIC.name:
+            filters = [DbCollection.visibility == CollectionVisibility.PUBLIC]
+        elif visibility == CollectionVisibility.PRIVATE.name:
+            filters = [DbCollection.visibility == CollectionVisibility.PRIVATE]
+
+        collection_objects = [dict(r) for r in session.query(cls.table).filter(*filters).all()]
+
+        # Select the collection_columns
+        collections = [cls._copy_and_transform_columns(collection_columns, r) for r in collection_objects]
+
+        # Select the link_columns and dataset_columns
+        for relation_name, rel_cols in (("links", link_columns), ("datasets", dataset_columns)):
+            for collection in collections:
+                if relation_name in collection:
+                    updated_relations = []
+                    for rel in collection[relation_name]:
+                        updated_relations.append(cls._copy_and_transform_columns(rel_cols, rel))
+                    collection[relation_name] = updated_relations
+
+        return collections
+
+    @staticmethod
+    def _copy_and_transform_columns(cols: typing.Iterable, from_obj: dict) -> dict:
```

Comment:
[to_dict](https://github.com/chanzuckerberg/single-cell-data-portal/blob/69505d5406796441556320c6a1fb9965d0735285/backend/corpora/common/corpora_orm.py#L61) does this not accomplish the same thing?